src/blob/index/core.rs

```diff
@@ -292,6 +292,20 @@ where
     }
 
     async fn get_all(&self, key: &K) -> Result<ReadResult<Vec<RecordHeader>>> {
+        let with_deletion = self.get_all_with_deletion_marker(key).await?;
+        Ok(with_deletion.map(|mut hs| {
+            if let Some(h) = hs.last() {
+                if h.is_deleted() {
+                    hs.truncate(hs.len() - 1);
+                }
+                hs
+            } else {
+                hs
+            }
+        }))
+    }
+
+    async fn get_all_with_deletion_marker(&self, key: &K) -> Result<ReadResult<Vec<RecordHeader>>> {
```

Comment:
Returning `ReadResult<>` only makes things harder to process on upper level. I think it is better to return just `Result<Vec<RecordHeader>>`

Reply:
Indeed, thank you